api/database_cluster_restore.go

```diff
@@ -49,7 +54,33 @@ func (e *EverestServer) ListDatabaseClusterRestores(ctx echo.Context, namespace,
 	path = strings.TrimSuffix(path, name)
 	path = strings.ReplaceAll(path, "database-clusters", "database-cluster-restores")
 	req.URL.Path = path
-	return e.proxyKubernetes(ctx, namespace, databaseClusterRestoreKind, "")
+
+	user, err := rbac.GetUser(ctx)
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, Error{
+			Message: pointer.ToString("Failed to get user from context" + err.Error()),
+		})
+	}
+	rbacFilter := transformK8sList(func(l *unstructured.UnstructuredList) error {
+		allowed := []unstructured.Unstructured{}
+		for _, obj := range l.Items {
+			restore := &everestv1alpha1.DatabaseClusterRestore{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, restore); err != nil {
+				e.l.Error(errors.Join(err, errors.New("failed to convert unstructured to DatabaseClusterRestore")))
+				return err
+			}
+			if err := e.enforceDBClusterListRestoreRBAC(user, restore); errors.Is(err, errInsufficientPermissions) {
```

Comment:
just a nitpick, but is a separate method actually needed? I see it calls just one `e.enforce()` internally, so I'm not sure its actually useful

Reply:
It's not necessary per se, I just wanted to be consistent with the rest of the codebase.
I think it hides unnecessary logic well in this case - it's easy to understand what the line does without extra error handling, logging, etc.